move/axelar_gateway/sources/types/message_status.move

```diff
@@ -0,0 +1,26 @@
+module axelar_gateway::message_status;
+
+use axelar_gateway::bytes32::Bytes32;
+
+// -----
+// Types
+// -----
+/// The Status of the message.
+/// Can be either one of two statuses:
+/// - Approved: Set to the hash of the message
+/// - Executed: Message was already executed
+public enum MessageStatus has copy, drop, store {
+    Approved(Bytes32),
+    Executed,
+}
+
+// -----------------
+// Package Functions
+// -----------------
+public (package) fun approved(hash: Bytes32): MessageStatus {
+    MessageStatus::Approved(hash)
+}
+
+public (package) fun executed(): MessageStatus {
```

Comment:
If you apply the suggestion around data and logic I mentioned, then I'd prefer to define this in the V0 data module so it's easier to work with the enum directly (e.g. if we want to use `match`).